RealmSwift/Object.swift

```diff
@@ -273,12 +273,15 @@ open class Object: RLMObjectBase, ThreadConfined {
      Returns whether two Realm objects are equal.
 
      Objects are considered equal if and only if they are both managed by the same Realm and point to the same
-     underlying object in the database.
+     underlying object in the database. Object comparison only takes place if the object type has a primary key.
```

Comment:
This new sentence seems incorrect, given the implementation below. `RLMObjectBaseAreEqual` doesn't care if objects have primary keys.